service/quota.py

```diff
@@ -1,16 +1,106 @@
 from threepio import logger
 
-from core.models import IdentityMembership, Identity, Provider
-
+from core.models import IdentityMembership, Identity
+from core.models.quota import (
+    has_floating_ip_count_quota,
+    has_port_count_quota,
+    has_instance_count_quota,
+    has_cpu_quota,
+    has_mem_quota,
+    has_storage_quota,
+    has_storage_count_quota,
+    has_snapshot_count_quota
+    )
 from service.cache import get_cached_driver
 from service.driver import get_account_driver
 
+
+def check_over_instance_quota(
+        username, identity_uuid,
+        esh_size=None, action=None):
+    """
+    Checks quota based on current limits (and an instance of size, if passed).
+
+    return or raise exc
+    """
```

Comment:
esh was a 'shorthand' for what could be better described as `cloud_size` or more verbosely `cloud_representation_of_size`.